packages/velo-external-db-core/lib/router.js

```diff
@@ -26,27 +28,36 @@ const initServices = (_schemaAwareDataService, _schemaService, _operationService
     filterTransformer = _filterTransformer
     aggregationTransformer = _aggregationTransformer
     roleAuthorizationService = _roleAuthorizationService
-    hooks = _hooks || {}
+    dataHooks = _hooks?.dataHooks || {}
+    schemaHooks = _hooks?.schemaHooks || {}
 }
 
-const serviceContext = {
+const serviceContext = () => ({
     schemaAwareDataService,
     schemaService
-}
+})
 
 
-const executeHooksFor = async(action, payload, requestContext) => {
+const executeDataHooksFor = async(action, payload, requestContext) => {
+    let result = payload
```

Comment:
1. This line is bug prone: since `payload` is being passed by ref, if you change `result`(which you do) - `payload` will change as well.

2. In general, IMO `let` is a bad practice, because it makes it harder to understand what's in this variable in any point in time. a better way of doing this is to use `reduce` and setting the variable to a `const`



3. I would be more descriptive and call it something like `lastHookResult`

Reply:
2. I agree, but the reduce function does not work well with sequential asynchronous actions. If you have a more elegant solution, I would be glad to hear it.

Reply:
1,3 fixed